cpp/src/glm/qn/glm_logistic.cuh

```diff
@@ -41,7 +43,10 @@ struct LogisticLoss : GLMBase<T, LogisticLoss<T>> {
   }
 
   inline __device__ T dlz(const T y, const T z) const {
-    return T(1.0) / (T(1.0) + MLCommon::myExp(-z)) - y;
+    // To avoid fp overflow with exp(z) when abs(z) is large
+    T ez = MLCommon::myExp(z < 0 ? z : -z);
+    T sigmoid = z < 0 ? ez / (T(1.0) + ez) : T(1.0) / (T(1.0) + ez);
```

Comment:
division is also a costly operation. So, can you also make sure that division operation doesn't appear in both if-else branches, just like it is being done for `myExp` above?

Reply:
Thanks, I have modified it.